demos/sahi/src/utils.py

```diff
@@ -0,0 +1,104 @@
+import argparse
+import urllib.request
+from os import path
+from pathlib import Path
+
+import torch
+from sahi.model import Yolov5DetectionModel
+
+
+def download_yolov5_model(model_url: str, destination_path: str):
+    Path(destination_path).parent.mkdir(parents=True, exist_ok=True)
+
+    if not path.exists(destination_path):
+        urllib.request.urlretrieve(
+            model_url,
+            destination_path,
+        )
+
+
+def obtain_detection_model(confidence_threshold: float):
+    device = (
+        f"cuda:{torch.cuda.current_device()}" if torch.cuda.is_available() else "cpu"
+    )
+    yolov5_model_path = "./models/yolov5x6.pt"
+    download_yolov5_model(
+        model_url="https://github.com/ultralytics/yolov5/releases/download/v5.0/yolov5x6.pt",
+        destination_path=yolov5_model_path,
+    )
+    return Yolov5DetectionModel(
+        model_path=yolov5_model_path,
+        confidence_threshold=confidence_threshold,
+        device=device,
+    )
+
+
+def create_arg_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(description="Track objects using sahi in a video.")
+    parser.add_argument("file", type=str, help="Video files to process")
+    parser.add_argument(
+        "--output-path", type=str, default="output.mp4", help="Output video path"
+    )
+    parser.add_argument(
+        "--distance-threshold",
+        type=float,
+        help="Norfair's distance threshold",
+        default=0.7,
+    )
+    parser.add_argument(
+        "--skip-period",
+        type=int,
+        help="Norfair's skip period",
+        default=1,
+    )
+    parser.add_argument(
+        "--initialization-delay",
+        type=int,
+        help="Norfair's initialization delay",
+        default=15,
+    )
+    parser.add_argument(
+        "--hit-counter-max",
+        type=int,
+        help="Norfair's hit counter max",
+        default=30,
+    )
+    parser.add_argument(
+        "--disable-sahi",
+        dest="enable_sahi",
+        help="Disable SAHI implementation, run predictions in the whole video",
+        action="store_false",
+        default=True,
+    )
+    parser.add_argument(
+        "--slice-height",
+        type=int,
+        help="Sahi's slice height",
+        default=256,
+    )
+    parser.add_argument(
+        "--slice-width",
+        type=int,
+        help="Sahi's slice width",
+        default=256,
+    )
+    parser.add_argument(
+        "--overlap-height-ratio",
+        type=float,
+        help="Sahi's overlap height ratio",
+        default=0.2,
+    )
+    parser.add_argument(
+        "--overlap-width-ratio",
```

Comment:
To keep things simpler, we can use just one `--overlap-ratio` argument instead of two `--overlap-width-ratio` and `--overlap-height-ratio` and set both `overlap_width_ratio` and `overlap_height_ratio` as `overlap_ratio`.

Reply:
I love this suggestion too, I tackled it to make everything simpler!